crates/bevy_gizmos/src/lib.rs

```diff
@@ -232,15 +232,21 @@ fn draw_all_aabbs(
     }
 }
 
+fn gold_kronecker(bits: u64) -> f32 {
+    // from https://extremelearning.com.au/unreasonable-effectiveness-of-quasirandom-sequences/
+    //
+    // See https://en.wikipedia.org/wiki/Low-discrepancy_sequence
+    // Map a sequence of integers (eg: 154, 155, 156, 157, 158) into the [0.0..1.0] range,
+    // so that the closer the numbers are, the larger the difference of their image.
+
+    // (u64::MAX / Φ) rounded down
+    // see https://probablydance.com/2018/06/16/fibonacci-hashing-the-optimization-that-the-world-forgot-or-a-better-alternative-to-integer-modulo/
+    const FRAC_U64MAX_GOLDEN_RATIO: u64 = 11400714819323198485;
+    const RATIO_360: f32 = u64::MAX as f32 / 360.0;
+    bits.wrapping_mul(FRAC_U64MAX_GOLDEN_RATIO) as f32 / RATIO_360
+}
 fn color_from_entity(entity: Entity) -> Color {
-    use bevy_utils::RandomState;
-    const U64_TO_DEGREES: f32 = 360.0 / u64::MAX as f32;
-    const STATE: RandomState =
-        RandomState::with_seeds(5952553601252303067, 16866614500153072625, 0, 0);
-
-    let hash = STATE.hash_one(entity);
-    let hue = hash as f32 * U64_TO_DEGREES;
-    Color::hsl(hue, 1., 0.5)
+    Color::hsl(gold_kronecker(entity.to_bits()), 1., 0.5)
```

Comment:
Just a quick comment: there are no guarantees of the layout of `entity.to_bits()`. Right now it's 32-bit generation appended with 32-bit index, but could switch to something like interleaved bits to ensure the numeric values stay small. Since there shouldn't be gizmos drawn for any entities which have been despawned, `entity.index()` values should not have any duplicates. Would it be better and more stable to just use `entity.index()` instead?

Reply:
Yeah. It seems fine to use `index()` instead. On a purely performance aspect, this becomes basically a noop. Great suggestion.